File: backend/src/data/models/Service.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ServiceModel = void 0;
const mongoose_1 = __importDefault(require("mongoose"));
const ServiceSchema = new mongoose_1.default.Schema({
    name: { type: String, required: true },
    price: { type: Number, required: true },
    description: { type: String },
    available: { type: Boolean, default: true },
});
exports.ServiceModel = mongoose_1.default.model('Service', ServiceSchema);

File: frontend/lib/data/models/service.dart
import 'dart:ffi';

class Service {
  final String name;
  final Float price;
  final String? description;
  final bool available;

  Service({
    required this.name,
    required this.price,
    required this.description,
    required this.available,
  });

  factory Service.fromJson(Map<String, dynamic> json) {
    return Service(
      name: json['name'] ?? '',
      price: json['price'] ?? 0.0,
      description: json['description'] ?? '',
      available: json['available'] ?? false,
    );
  }



  Map<String, dynamic> toJson() {
    return {
      'name': name,
      'price': price,
      'description': description,
      'available': available,
    };
  }

  Service copyWith({
    String? name,
    Float? price ,
    String? description,
    String? available,
  }) {
    return Service(
      name: this.name,
      price: this.price,
      description: description ?? this.description,
      available: this.available,
    );
  }
}

